kornia/augmentation/augmentations.py

```diff
@@ -0,0 +1,70 @@
+import torch
+import torch.nn as nn
+from typing import Tuple
+from kornia.geometry.transform.flips import hflip
+
+
+class RandomHorizontalFlip(nn.Module):
+
+    r"""Horizontally flip a tensor image or a batch of tensor images randomly with a given probability.
+    Input should be a tensor of shape (C, H, W) or a batch of tensors :math:`(*, C, H, W)`.
+
+    Args:
+        p (float): probability of the image being flipped. Default value is 0.5
+    """
+
+    def __init__(self, p=0.5):
+        super(RandomHorizontalFlip, self).__init__()
+        self.p: float = p
+
+    def __repr__(self):
+        repr = "(p={0})".format(self.p)
+        return self.__class__.__name__ + repr
+
+    def forward(self, input: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:  # type: ignore
+        return random_hflip(input, self.p)
+
+
+def random_hflip(input, p=0.5) -> Tuple[torch.Tensor, torch.Tensor]:
+
+    r"""Horizontally flip a tensor image or a batch of tensor images randomly with a given probability.
+    Input should be a tensor of shape (C, H, W) or a batch of tensors :math:`(*, C, H, W)`.
+
+    Args:
+        p (float): probability of the image being flipped. Default value is 0.5
+    Returns:
+        torch.Tensor: The horizontally flipped input
+        torch.Tensor: The applied transformation matrix
```

Comment:
update the description explaining that depending on the returning flag can be tuple or single raw tensor